Cython/Compiler/Options.py

```diff
@@ -346,6 +347,21 @@ class DEFER_ANALYSIS_OF_ARGUMENTS:
     pass
 DEFER_ANALYSIS_OF_ARGUMENTS = DEFER_ANALYSIS_OF_ARGUMENTS()
 
+class BoundsCheckEnum(enum.Enum):
```

Comment:
I don't know how many future uses we'll have for this. It's obviously can't be used where nodes would be transformed differently.

My idea is that all existing code that expects true/false can continue to work, but we have the option of looking for the special case if we want.

Reply:
It looks like your "thread safety" option came back in disguise. I'd rather consider thread safety (or not) a side effect here, rather than something that we explicitly code into `boundscheck`.